src/components/Movie/Movie.js

```diff
@@ -1,25 +1,61 @@
-import React from "react";
+import React, {Component} from "react";
 import PropTypes from 'prop-types';
 import './Movie.css';
 import Hero from "../Hero/Hero";
 import Media from "../Media/Media";
 import MovieDescription from "../MovieDescription/MovieDescription";
+import { NavLink } from 'react-router-dom';
 
-const Movie = ({data, clearSelectedMovie, videos}) => {
-    return (
-        <section className='Movie'>
-            <button onClick={clearSelectedMovie}>Go Back</button>
-            <Hero info={data}/>
-            <MovieDescription info={data} />
-            {videos && <Media videos={videos} />}
-        </section>
-    )
+class Movie extends Component {
+    constructor() {
+        super();
+        this.state = {
+            selectedMovie: null,
```

Comment:
i feel much better about movie keeping track of its own state. seems like it keeps issues at bay more too